src/DataStructures/Tensor/EagerMath/DotProduct.hpp

```diff
@@ -0,0 +1,65 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+/// \file
+/// Defines functions euclidean dot_product and dot_product with a metric
+
+#pragma once
+
+#include "DataStructures/DataVector.hpp"
+#include "DataStructures/Tensor/Tensor.hpp"
+#include "ErrorHandling/Assert.hpp"
+#include "Utilities/MakeWithValue.hpp"
+
+/*!
+ * \ingroup TensorGroup
+ * \brief compute the Euclidean dot product of two rank-1 tensors
+ *
+ * \returns the Euclidean dot product of the rank-1 tensors
+ *
+ * \details
+ * Returns \f$A^a B^b \delta_{ab}\f$ for input tensors \f$A^a\f$ and \f$B^b\f$.
+ */
+template <typename DataType, typename Index>
+Scalar<DataType> dot_product(
+    const Tensor<DataType, Symmetry<1>, tmpl::list<Index>>& tensor_a,
```

Comment:
I'd prefer `vector_a` etc. How do others feel about this?

Reply:
I like vector_a better, given this is really an operation applicable only to rank-1 tensors (vectors and one-forms). It's only called tensor_a because I wanted to be consistent with magnitude, which calls the input tensor.